.mk/development.mk

```diff
@@ -69,3 +69,23 @@ undeploy-infra: undeploy-loki undeploy-grafana uninstall
 
 .PHONY: undeploy-all
 undeploy-all: undeploy-infra undeploy-sample-cr undeploy-sample-workload
+
+.PHONY: deploy-prometheus
+deploy-prometheus: ## Deploy prometheus.
+	@echo -e "\n==> Deploy prometheus"
+	@echo -e "\n==> Obtain flp node addr/ports"
+	pod_list=$$(kubectl get pods --selector=app=flowlogs-pipeline -o jsonpath='{.items[*].status.podIP}'); \
+	FLP_TARGET_LIST=$$(echo $$pod_list | sed 's~^~[~; s~ ~:9102,~g; s~$$~:9102]~;') ; \
+	sed "s~%FLP_TARGETS%~$$FLP_TARGET_LIST~" config/kubernetes/deployment-prometheus.yaml >/tmp/deployment-prometheus.yaml
```

Comment:
this can be ok as a temporary deploy script, but then we should switch to something more robust using `kubernetes_sd_config` (or alternatively, a ServiceMonitor with prom operator). The problem with listing static IPs is that we'll need to reconfigure prometheus every time the FLP pods restart (e.g. after a config change). That will be inconvenient.

Here's a typical prom config to scrape kube pods based on annotations:
```yaml

  - job_name: "kubernetes-pods"

    kubernetes_sd_configs:
      - role: pod

    relabel_configs:
      # Example relabel to scrape only pods that have
      # "prometheus.io/scrape = true" annotation.
      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scrape]
        action: keep
        regex: true
      # Example relabel to customize metric path based on pod
      # "prometheus.io/metric_path = <metric path>" annotation.
      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_metric_path]
        action: replace
        target_label: __metrics_path__
        regex: (.+)
      # Example relabel to scrape only single, desired port for the pod
      # based on pod "prometheus.io/scrape_port = <port>" annotation.
      - source_labels: [__address__, __meta_kubernetes_pod_annotation_prometheus_io_scrape_port]
        action: replace
        regex: ([^:]+)(?::\d+)?;(\d+)
        replacement: $1:$2
        target_label: __address__
      - action: labelmap
        regex: __meta_kubernetes_pod_label_(.+)
      - source_labels: [__meta_kubernetes_namespace]
        action: replace
        target_label: namespace
      - source_labels: [__meta_kubernetes_pod_name]
        action: replace
        target_label: pod
```

But, maybe we don't want prom to scrape any pod in the cluster either. So we can also be more selective and list explicitly the pods we want (e.g. we can filter by app label)

Reply:
Changed to use kubernetes_sd_configs.